pkg/services/alerting/ticker.go

```diff
@@ -1,76 +1,64 @@
 package alerting
 
 import (
+	"fmt"
 	"time"
 
 	"github.com/benbjohnson/clock"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 )
 
 // Ticker is a ticker to power the alerting scheduler. it's like a time.Ticker, except:
 // * it doesn't drop ticks for slow receivers, rather, it queues up.  so that callers are in control to instrument what's going on.
-// * it automatically ticks every second, which is the right thing in our current design
-// * it ticks on intervalSec marks or very shortly after. this provides a predictable load pattern
+// * it ticks on interval marks or very shortly after. this provides a predictable load pattern
 //   (this shouldn't cause too much load contention issues because the next steps in the pipeline just process at their own pace)
 // * the timestamps are used to mark "last datapoint to query for" and as such, are a configurable amount of seconds in the past
-// * because we want to allow:
-//   - a clean "resume where we left off" and "don't yield ticks we already did"
-//   - adjusting offset over time to compensate for storage backing up or getting fast and providing lower latency
-//   you specify a lastProcessed timestamp as well as an offset at creation, or runtime
 type Ticker struct {
 	C           chan time.Time
 	clock       clock.Clock
 	last        time.Time
-	offset      time.Duration
-	newOffset   chan time.Duration
-	intervalSec int64
-	paused      bool
+	interval    time.Duration
+	behindTicks prometheus.Gauge
 }
 
-// NewTicker returns a ticker that ticks on intervalSec marks or very shortly after, and never drops ticks
-func NewTicker(last time.Time, initialOffset time.Duration, c clock.Clock, intervalSec int64) *Ticker {
+// NewTicker returns a ticker that ticks on interval marks or very shortly after, and never drops ticks. interval should not be negative or zero
+func NewTicker(c clock.Clock, interval time.Duration) *Ticker {
+	if interval <= 0 {
+		panic(fmt.Errorf("non-positive interval [%v] is not allowed", interval))
```

Comment:
In the case of very small durations, the worst-case scenario is that it will fire at the pace of the consumer, which is fine from the perspective of the timer itself.

Reply:
In such cases the ticker will send timestamps in the past as we don't drop ticks. If the consumer cannot consume at the rate that the timer fires then the timer will start to fall further and further behind wall clock time and never catch up.